Added control code for LED panel

kohlerz opened the pull request:
Created a ROS node for the led panel, as well as a service. There is also Arduino code to convert the serial output to LED commands

bennowotny commented on src/wr_led_matrix/README.md:
```diff
@@ -0,0 +1,15 @@
+# Changes
```
This file was just instructions for making the package, it doesn't really need to be in the final project.

bennowotny replied:
This file is removed.

bennowotny commented on src/wr_led_matrix/package.xml:
```diff
@@ -0,0 +1,61 @@
+  <!-- One maintainer tag required, multiple allowed, one person per tag -->
+  <!-- Example:  -->
+  <!-- <maintainer email="jane.doe@example.com">Jane Doe</maintainer> -->
+  <maintainer email="parallels@todo.todo">parallels</maintainer>
```
Maybe throw your name in here just to know you were the main developer.  Otherwise, you might not be the first person we would go to with questions.

bennowotny replied:
This is still not changed, which may be okay, but you should comment saying that you want it this way to resolve this.

kohlerz replied:
I must have forgot to save the file, but the new commit has my email and github username

bennowotny commented on src/wr_led_matrix/src/LEDMatrix.py:
```diff
@@ -0,0 +1,36 @@
+from wr_led_matrix.srv import LEDMatrix,LEDMatrixRequest,LEDMatrixResponse
+
+s: serial.Serial
+baud = rospy.get_param("/led_matrix/baud")
+port = rospy.get_param("/led_matrix/com_port")
```
These should use a private namespace reference since the name of the node could change based on what's in the launch file.

bennowotny replied:
This change has been addressed.

bennowotny reviewed (changes requested):
Some minor changes logged in comments.  Additionally, we should add the `settings.json` file in here for consistency in linting and some project settings, so we should update the `gitignore` file to permit the settings file as well and be conscious to keep it pruned.

bennowotny commented:
Also, I found that pySerial has a way to disable DTR in their [constructor](https://pyserial.readthedocs.io/en/latest/pyserial_api.html#serial.Serial), which should prevent the Arduino from resetting.  Could we try opening/closing the `Serial` object in the callback with DTR disabled?

bennowotny commented:
With respect to the linting question, don't make changes or include extra linting files in this PR; this will be handled in #25.

kohlerz commented:
I just updated everything you asked for. The settings.json file should be usable by everyone, and I excluded it from the gitignore. I also renamed some of the files for consistency

bennowotny commented on .gitignore:
```diff
@@ -9,6 +9,7 @@ devel/
 /.vscode/*
 # except recommended extensions
 !/.vscode/extensions.json
+~/.vscode/settings.json
```
This change can be reverted, due to its relation to linting, which is handled by #25 (the line can just be removed).

kohlerz replied:
I removed this line from the gitignore, sorry for the confusion

kohlerz replied:
I don't think I can resolve this one myself, but if I'm wrong let me know

bennowotny reviewed (changes requested):
In general, looks good, just one change wrt [this](https://github.com/WisconsinRobotics/WRover_Software/pull/24#issuecomment-1126830300) comment.  Otherwise, comment on the notes left so I know they were addressed, since each conversation has to be marked 'resolved' before the merge can happen.

bennowotny reviewed (approved):
Good to go!  In the future, don't resolve conversations you didn't open, just say in a reply that you should have fixed it (Maybe include a commit hash/link).  Then, the person that opened that conversation can verify that it was satisfied prior to 'resolving' the conversation.

Not a big deal; this looks good.  Thanks!
